---
title: "About Data"
---

We have the raw reads fastq files from two different sets of sequencing. The first set of data is comparison between Methyl-Cap and TAYN-Cap. The methyl cap is data is the wild type and TAYNCap represents the mutant type reader developed by the lab. Methyl-CpG binding domain-based capture and sequencing (MethylCap , MBD-Seq , MBDCap-Seq , MBDCap) or MBD-isolated genome sequencing (MiGS ), uses proteins to capture methylated DNA in the genome. Genomic DNA is first sonicated and incubated with tagged MBD proteins that can bind methylated cytosines. The protein-DNA complex is then precipitated with antibody-conjugated beads that are specific to the protein tag. Deep sequencing provides greater genome coverage, representing the majority of MBD-bound methylated DNA.

In addition to this, MeDipSeq and hMeDipSeq data is also present. Methylated DNA immunoprecipitation sequencing (MeDIP-Seq), DNA immunoprecipitation sequencing (DIP-Seq) or Hydroxymethylated DNA immunoprecipitation sequencing (hMeDIP-Seq), is commonly used to study 5mC or 5hmC modification. Specific antibodies can be used to study cytosine modifications. If using 5mC-specific antibodies, methylated DNA is isolated from genomic DNA via immunoprecipitation. Anti-5mC antibodies are incubated with fragmented genomic DNA and precipitated, followed by DNA purification and sequencing. Deep sequencing provides greater genome coverage, representing the majority of immunoprecipitated methylated DNA.