---
title: "Analysis of MeDip-Seq data"
author: "Shashank Tiwari"
date: "Most recent update:<br> `r Sys.Date()`"
output: 
  rmarkdown::html_document: 
    theme: spacelab
    highlight: zenburn 
    code_folding: show 
    toc: true 
    toc_float: true
    smooth_scroll: true
    number_sections: false 
    self_contained: true 
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=T, message=F}
#root.dir <- here::here()
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  #root.dir = root.dir
  fig.height = 6,
  fig.width = 7.00787 #178mm
)  
knitr::opts_knit$set(#root.dir = root.dir, 
                     dpi = 350)  
library(data.table)
library(ggplot2)
library(tidyverse)
library(cowplot)
library(viridis)
library(dplyr)
library(genomation)
library(biomaRt)
library(GenomicRanges)
library(DT)
library(magick)
library(patchwork)
library(slickR)
library(svglite)
#library(ggpubr)
```



1. TAYN and WT consensus peak: 5946peaks
2. hMeDip and MeDip consensus peak: 8760peaks
3. DMRs from MEDIPS: 23607peaks
4. DMRs intersected with MBDSeq consensus peaks: 1911peaks
5. DMRs intersected with WT negative and TAYN positive peaks: 1205peaks


## Annotating the peak regions
I used the annotatePeaks tool from `homer` to annotate the enriched regions and then used Rscript to generate the plots. To annotate the location of a given peak in terms of important genomic features, annotatePeaks.pl calls a separate program (assignGenomeAnnotation) to efficiently assign peaks to one of millions of possible annotations genome wide. Two types of output are provided. The first is "Basic Annotation" that includes whether a peak is in the TSS (transcription start site), TTS (transcription termination site), Exon (Coding), 5' UTR Exon, 3' UTR Exon, Intronic, or Intergenic, which are common annotations that many researchers are interested in. A second round of "Detailed Annotation" also includes more detailed annotation, also considering repeat elements and CpG islands. Since some annotation overlap, a priority is assign based on the following (in case of ties it's random [i.e. if there are two overlapping repeat element annotations]): TSS (by default defined from -1kb to +100bp) TTS (by default defined from -100 bp to +1kb) CDS Exons 5' UTR Exons 3' UTR Exons".

## DMRs intersected with MBDSeq consensus peaks

The regions of interest are generated by the following steps

* Peaks called for all conditions using MACS2
* Consensus Peaks are generated for MeDip-Seq data.
* MEDIPS is used to identify Differentialy methylated regions between TAYNCap-Seq and MethyCap-Seq.
* DMRs from MEDIPS are intersected with peaks which are present in both TAYNCap-Seq and MethylCap-Seq.


These are the regions which are enriched in TAYN and Methyl-Seq and TAYN > WT.


```{r, echo=FALSE, out.width="49%", fig.align='center',fig.show='hold',fig.cap="DMRs intersected with DipSeq and CapSeq consensus peaks"}
knitr::include_graphics(c("Images/Annotate/DMR_MBD_and_TAYN_repeat/DMRs_MBD_consensus.png","Images/Annotate/DMR_MBD_and_TAYN_repeat/detailed_DMRs_MBD_consensus.png"))
#knitr::include_graphics("~/Pictures/RBPFOX2.png")
```



Here is a table of CG and CA percentages of reads. Note that here the sequences in lowercase represent softmasked repeat regions and uppercase
letters represent non-repeats sequence. As you can see majority of reads here come from repeat regions.

```{r, echo=FALSE}
df <- data.table(fread("CG_DMRs_MBD_consensus.tsv"))
DT::datatable(df[,c(1:3,5:9,4)], options = list(scrollX=TRUE))

```



## DMR intersected with only TAYN positive and WT negative peaks

The regions of interest are generated by the following steps:

*   Peaks called for all conditions using MACS2
*   Peaks from TAYN and WT are intersected to generate regions only present in TAYN.
*   MEDIPS is used to identify Differentialy methylated regions between TAYNCap-Seq and MethyCap-Seq.
*   DMRs from MEDIPS are intersected with peaks which are present in TAYNCapSeq but not present in MethylCap-Seq data.

These are the regions which are not enriched in WT/Methyl-Seq and are enriched in the Mutant/TAYN-Seq data.


```{r, out.width="49%", echo=FALSE, fig.show='hold', fig.cap="DMRs intersected with Dip consensus peaks only", fig.align='center'}
knitr::include_graphics(c("Images/Annotate/DMR_MBD_and_TAYN_repeat/DMRs_TAYN_pos_wt_neg.png","Images/Annotate/DMR_MBD_and_TAYN_repeat/detailed_DMRs_TAYN_pos_wt_neg.png"))
```


Here is a table of CG and CA percentages of reads.

```{r, echo=FALSE}
df1 <- data.table(fread("CG_DMRs_TAYN_pos_wt_neg.fasta"))
DT::datatable(df1[,c(1:3,5:9,4)], options = list(scrollX=TRUE))

```


